src/usb.ts

```diff
@@ -199,15 +199,23 @@ export class USB extends EventDispatcher {
                 }
 
                 function selectFn(device: USBDevice) {
+                    if (!(device instanceof USBDevice)) {
+                        return reject(new Error("requestDevice error: invalid device type."));
+                    }
+
                     if (!this.replaceAllowedDevice(device)) this.allowedDevices.push(device);
                     resolve(device);
                 }
 
                 // If no devicesFound function, select the first device found
                 if (!this.devicesFound) return selectFn.call(this, devices[0]);
 
-                const selectedDevice = this.devicesFound(devices, selectFn.bind(this));
-                if (selectedDevice) selectFn.call(this, selectedDevice);
+                return this.devicesFound(devices)
+                .then(device => {
+                    if (device) {
```

Comment:
Maybe we should throw an error if a device wasn't selected?

Reply:
yep, good call.